crates/turborepo-lib/src/diagnostics.rs

```diff
@@ -0,0 +1,523 @@
+        chan.send(DiagnosticMessage::NotApplicable(
+            Self::NAME.to_string(),
+            message.to_string(),
+        ))
+        .await
+        .ok(); // channel closed, ignore
+    }
+}
+
+pub struct GitDaemonDiagnostic;
+
+impl Diagnostic for GitDaemonDiagnostic {
+    const NAME: &'static str = "git.daemon";
+
+    fn execute(&self, chan: MpscSender<DiagnosticMessage>) {
+        tokio::task::spawn(async move {
+            if !(cfg!(target_os = "windows") || cfg!(target_os = "macos")) {
+                // the git daemon is not implemented on unix
+                Self::not_applicable(chan, "Git FS Monitor (not available on unix)".to_string())
+                    .await;
+                return;
+            }
+
+            Self::started(&chan, "Git FS Monitor".to_string()).await;
+
+            let futures: Result<Vec<Vec<u8>>, _> = join_all(
+                [
+                    &["--version"][..],
+                    &["config", "--get", "core.fsmonitor"][..],
+                    &["config", "--get", "core.untrackedcache"][..],
+                ]
+                .into_iter()
+                .map(|args| async move {
+                    // get the current setting
+                    let stdout = Stdio::piped();
+
+                    let command = Command::new("git")
+                        .args(args)
+                        .stdout(stdout)
+                        .spawn()
+                        .expect("too many processes"); // this can only fail if we run out of processes on unix
+
+                    command.wait_with_output().await.map(|d| d.stdout)
+                }),
+            )
+            .await
+            .into_iter()
+            .collect(); // transpose
+
+            Self::log_line(&chan, "Collecting metadata".to_string()).await;
+
+            match futures.as_ref().map(|v| v.as_slice()) {
+                Ok([version, fsmonitor, untrackedcache]) => {
+                    let version = String::from_utf8_lossy(version);
+                    let Some(version) = version.trim().strip_prefix("git version ") else {
+                        Self::failed(chan, "Failed to get git version".to_string()).await;
+                        return;
+                    };
+
+                    let (major, rest) = version.split_once('.').expect("semver");
+                    let (minor, _) = rest.split_once('.').expect("semver");
+
+                    let major = major.parse::<u32>().expect("int");
+                    let minor = minor.parse::<u32>().expect("int");
+
+                    if major == 2 && minor < 37 || major == 1 {
+                        Self::failed(
+                            chan,
+                            format!(
+                                "Git version {} is too old, please upgrade to 2.37 or newer",
+                                version
+                            ),
+                        )
+                        .await;
+                        return;
```

Comment:
We already use the [semver](https://docs.rs/semver/latest/semver/) crate and that might be a little more robust than handrolling parsing and range checks.